src/limb.rs

```diff
@@ -69,6 +69,42 @@ pub fn limbs_less_than_limbs_consttime(a: &[Limb], b: &[Limb]) -> LimbMask {
     unsafe { LIMBS_less_than(a.as_ptr(), b.as_ptr(), b.len()) }
 }
 
+#[cfg(feature = "rsa_keygen")]
+pub fn limbs_sub_limb(r: &mut [Limb], a: &[Limb], b: Limb) {
+    extern "C" {
+        fn LIMBS_sub_limb(r: *mut Limb, a: *const Limb, b: Limb, num_limbs: c::size_t);
+    }
+    assert_eq!(r.len(), a.len());
+    unsafe { LIMBS_sub_limb(r.as_mut_ptr(), a.as_ptr(), b, a.len()) }
+}
+
+#[cfg(feature = "rsa_keygen")]
+pub fn limbs_count_low_zero_bits(v: &[Limb]) -> u16 {
+    extern "C" {
+        fn BN_count_low_zero_bits(bn: *const Limb, num_limbs: c::size_t) -> c::int;
+    }
+    let r = unsafe { BN_count_low_zero_bits(v.as_ptr(), v.len()) };
+    r as u16
+}
+
+#[cfg(feature = "rsa_keygen")]
+pub fn limbs_copy(r: &mut [Limb], a: &[Limb]) {
```

Comment:
You can do the copying in pure Rust code. (In *ring*, we can copy secret values from one place to another in Rust, we just don't do any arithmetic in Rust get.)